File: typescript/problems/1716.计算力扣银行的钱.ts
/*
 * @lc app=leetcode.cn id=1716 lang=typescript
 *
 * [1716] 计算力扣银行的钱
 */

// @lc code=start
// 模拟
function totalMoney(n: number): number {
  let ans = 0
  // 1 + ~~((i - 1) / 7) 每周一的钱
  // (i - 1) % 7 每周的第二天开始加钱
  for (let i = 1; i <= n; i++) ans += 1 + ~~((i - 1) / 7) + ((i - 1) % 7)
  return ans
}
// @lc code=end

File: python3/problems/1716.计算力扣银行的钱.py
#
# @lc app=leetcode.cn id=1716 lang=python3
#
# [1716] 计算力扣银行的钱
#

# @lc code=start
class Solution:
    def totalMoney(self, n: int) -> int:
        ans, wk, day = 0, 1, 1
        for _ in range(n):
            ans += wk + day - 1
            day += 1
            if day == 8:
                day = 1
                wk += 1
        return ans


# @lc code=end
